app/src/store/loginAndRegister.ts

```diff
@@ -0,0 +1,80 @@
+import { createSlice, type SerializedError } from '@reduxjs/toolkit';
+import { match } from 'ts-pattern';
+import { getAccessToken } from './voice/accessToken';
+import { register } from './voice/registration';
+import { type AsyncStoreSlice } from './app';
+import { createTypedAsyncThunk } from './common';
+import { login, logout } from './user';
+
+export type LoginAndRegisterRejectValue =
+  | {
+      reason: 'LOGIN_REJECTED';
+    }
+  | {
+      reason: 'GET_ACCESS_TOKEN_REJECTED';
+    }
+  | {
+      reason: 'REGISTER_REJECTED';
+    };
+
+export const loginAndRegister = createTypedAsyncThunk<
+  void,
+  void,
+  {
+    rejectValue: LoginAndRegisterRejectValue;
+  }
+>('registration/loginAndRegister', async (_, { dispatch, rejectWithValue }) => {
```

Comment:
Let's replicate the pattern we use in the other slice files, where we have a `const sliceName` and we use the `generateThunkActionTypes` function.